lib/server.js

```diff
@@ -336,9 +336,57 @@ async function create (log, error, config, routes, db, oauthdb, translator) {
   await server.register(require('hapi-fxa-oauth'))
 
   server.auth.strategy('oauthToken', 'fxa-oauth', config.oauth)
+
+  server.auth.scheme('fxa-oauth-refreshToken', (server, options) => {
+    return {
+      async authenticate (request, h) {
+        try {
+          const refreshTokenId = request.headers.authorization.split(' ')[1];
+          //const refreshTokenId = request.payload.refresh_token;
+          const refreshToken = await oauthdb.checkRefreshToken(refreshTokenId)
+          if (! refreshToken || ! refreshToken.active) {
+            return h.unauthenticated()
+          }
+
+          // Create a mock sessionToken that can be used in functions expect one.
+          const sessionToken = {
+            uid: refreshToken.sub,
+            id: refreshTokenId,
+            tokenVerified: true // TODO - not really sure about this.
+          }
+
+          console.log('refreshToken', refreshToken)
+
+          const devices = await db.devices(refreshToken.sub)
+          console.log('devices', devices, refreshTokenId)
+          const device = devices.filter(device => device.sessionToken === refreshTokenId)[0]
```

Comment:
After careful consideration, this conflation of `sessionToken` and `refreshTokenId` makes me just a little too uncomfortable.  I think it'd be worth attempting a DB migration to add a separate `refreshTokenId` column to the `devices` table, and seeing what flow-on effects this will have throughout the code.  It will make it harder for us to implement the "happy path" in the code here, but also much less likely to accidentally do the wrong thing.

One example: destroying the device record of an OAuth client, should also destroy the underlying refreshToken for that client. This seems harder to do right (and easier to do wrong) if we're overloading the existing sessionToken column rather than explicitly tracking a refreshToken column.

I'll try to dive into more of the details tomorrow.